nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,65 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/:id", async (request, response) => {
+  try {
+    // console.log(meals);
+    // console.log("in /api/meals");
+    const findId = meals.find((meal) => meal.id === Number(request.params.id));
+    if (findId) {
+      response.json(findId);
+    } else {
+      response.status(404).end("Invalid ID");
+    }
+  } catch (error) {
+    throw error;
+  }
+});
+
+// router.get("/", async (request, response) => {
+//   try {
+//     console.log(meals);
+//     console.log("in /api/meals");
+//     response.json(meals);
+//   } catch (error) {
+//     throw error;
+//   }
+// });
+
+router.get("/", async (request, response) => {
+  let filteredData = meals;
+  {
+    try {
+      if ("date" in request.query)
+        filteredData = filteredData.filter(
+          (meal) =>
+            new Date(meal.createdAt).getTime() >
+            new Date(request.query.date).getTime()
+        );
+    } catch (error) {
```

Comment:
This endpoint does not really work because you are looking for `date` but then trying to filter by `createdAt`. 

Good logic, just needs some adjustments to make it work 👍🏽

Reply:
Also, not sure, but you might need a `return` here to make it work, after adjusting variables,